### PR TITLE
Fix for NormalizePath resolving IntDir based on current working directory

### PR DESCRIPTION
**Pull request by aschultz**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/4108)

### CODE REVIEW

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 1]
```diff
@@ -50,7 +50,8 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
-    <UnmergedWinmdDirectory>$([MSBuild]::NormalizePath($(IntDir)UnmergedWinMD))</UnmergedWinmdDirectory>
+    <!-- For relative paths, make it relative to MSBuildProjectDirectory not CurrentWorkingDirectory -->
+    <UnmergedWinmdDirectory>$([MSBuild]::NormalizeDirectory( $([System.IO.Path]::Combine( $(MSBuildProjectDirectory), $(IntDir)UnmergedWinMD )) ))</UnmergedWinmdDirectory>
```
Comment (JunielKatarn) [inline vnext/ReactUWP/ReactUWP.vcxproj]: This looks more compact, but still not clear.
Where would the unmerged MD dir be located?
Being a derived artifact (though intermediate), it should go within the `IntDir` hierarchy.

Comment (aschultz) [reply vnext/ReactUWP/ReactUWP.vcxproj]: IntDir is a relative path, with the intent that it is relative to the project directory. However, NormalizeDirectory and NormalizePath resolve their inputs to an absolute path based on current working directory. Often times, that means C:\windows\system32.

For example:
IntDir = \build\x64\
NormalizedDir = c:\windows\system32\build\x64

Clearly that's no good. By using Path.Combine, I ensure the path is first resolved to absolute based on the project directory.

IntDir = \build\x64\
MSBuildProjectDir = c:\react-native-windows\vnext\reactcommon\
CombinedDir = c:\react-native-windows\vnext\reactcommon\build\x64
NormalizedDir =  c:\react-native-windows\vnext\reactcommon\build\x64\

Path.Combine is also smart -- if the second arg is already absolute, it just uses it as is.
Path.Combine("c:\foo", "d:\bar") == "d:\bar"

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 2]
```diff
@@ -50,7 +50,8 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
-    <UnmergedWinmdDirectory>$([MSBuild]::NormalizePath($(IntDir)UnmergedWinMD))</UnmergedWinmdDirectory>
+    <UnmergedWinmdDirectory>$([MSBuild]::NormalizePath( $(IntDir)UnmergedWinMD) ))</UnmergedWinmdDirectory>
+    <UnmergedWinmdDirectory>$([System.IO.Path]::Combine( $(ProjectDir), $(UnmergedWinmdDirectory) ))</UnmergedWinmdDirectory>
```
[the remark this answers is not in the document]
Comment (aschultz) [reply vnext/ReactUWP/ReactUWP.vcxproj]: My first revision was broken. Bad copy/pasting between RNW and my app's project. The latest iteration is my intent. See my other comment for why this change is necessary.

**Comment by aschultz** [comment]
@NickGerleman Mind merging this for me?

File: vnext/ReactUWP/ReactUWP.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply vnext/ReactUWP/ReactUWP.vcxproj]: I'm aware by default `IntDir` is something like `<thisdir>\obj\$(Platform)`.

In this case, we made `IntDir` NOT relative to the current directory.
See `vnext\Directory.Build.props`.

That said, as long as this doesn't mess with the intermediate artifact directory structure (`<root>\vnext\build\$(Platform)\$(Configuration)\$(ProjectName)`), this change should be fine.

**Pull request merged**